pageserver/src/tenant/tasks.rs

```diff
@@ -65,6 +65,11 @@ pub(crate) async fn concurrent_background_tasks_rate_limit_permit(
         .with_label_values(&[loop_kind.as_static_str()])
         .guard();
 
+    pausable_failpoint!(
+        "initial-size-calculation-permit-pause",
+        loop_kind == BackgroundLoopKind::InitialLogicalSizeCalculation
+    );
+
     match CONCURRENT_BACKGROUND_TASKS.acquire().await {
```

Comment:
I see your test uses `delay()`, so, it doesn't stall the executor thread by parking the thread, but, it still hogs the executor CPU.
Better to use `pause` with our async version: `libs/utils/src/failpoint_support.rs`

Reply:
The previous iteration of the test used `delay` with a vanilla fail point (maybe GH glitched and you saw the old version). It's now switched to a `pausable_failpoint` which sleeps in `spawn_blocking`. That's not ideal either, but it may be good enough for the purpose